### PR TITLE
EDSC-2623: Implements windowing and table view for granules panel

### PR DESCRIPTION
**Pull request by trevorlang**

### CODE REVIEW

File: static/src/js/components/EDSCTable/EDSCTable.js [hunk 1]
```diff
@@ -182,22 +246,45 @@ const EDSCTable = ({
...
+
       const { style: rowStyle, ...rowRest } = rowProps
 
+      const rowClasses = classNames([
+        'edsc-table__tr',
+        `edsc-table__tr--${(index + 1) % 2 === 0 ? 'even' : 'odd'}`,
+        ...rowClassesFromState
+      ])
+
+      const rowEvents = {
```
Comment (trevorlang) [inline static/src/js/components/EDSCTable/EDSCTable.js:273]: Add comment about events

File: static/src/js/components/GranuleResults/GranuleResultsHeader.js [hunk 2]
```diff
@@ -160,6 +172,20 @@ class GranuleResultsHeader extends Component {
     // Determine the correct granule count based on granules that have been removed
     const granuleCount = hits - excludedGranuleIds.length
 
+    const viewButtonListClasses = classNames([
+      'collection-results-header__view-button',
```
Comment (trevorlang) [inline static/src/js/components/GranuleResults/GranuleResultsHeader.js:176]: Update classname

File: static/src/js/components/GranuleResults/GranuleResultsItem.js [hunk 3]
```diff
@@ -1,144 +1,38 @@
 /* eslint-disable jsx-a11y/img-redundant-alt */
```
Comment (trevorlang) [inline static/src/js/components/GranuleResults/GranuleResultsItem.js:1]: Move above instance

File: static/src/js/components/GranuleResults/GranuleResultsListItem.scss [hunk 4]
```diff
@@ -0,0 +1,2 @@
+.granule-results-list-item {
```
Comment (trevorlang) [inline static/src/js/components/GranuleResults/GranuleResultsListItem.scss:1]: Remove

File: static/src/js/components/GranuleResults/GranuleResultsTableHeaderCell.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
+/* eslint-disable no-unused-vars */
```
Comment (trevorlang) [inline static/src/js/components/GranuleResults/GranuleResultsTableHeaderCell.js:1]: Not needed

File: static/src/js/components/GranuleResults/__tests__/GranuleResultsItem.test.js [hunk 6]
```diff
@@ -288,87 +298,78 @@ describe('GranuleResultsItem component', () => {
...
 
-      expect(downloadButton.props().dataLinks).toEqual([])
+      expect(downloadButton.length).toEqual(0)
     })
   })
 
-  describe('granule map events', () => {
-    test('hovering over a granule highlights the granule on the map', () => {
-      const { enzymeWrapper, props } = setup('cmr')
+  // describe('granule map events', () => {
```
Comment (trevorlang) [inline static/src/js/components/GranuleResults/__tests__/GranuleResultsItem.test.js:317]: Remove

File: static/src/js/components/GranuleResults/__tests__/GranuleResultsList.test.js [hunk 7]
```diff
@@ -1,182 +1,152 @@
...
-        const el = document.createElement('div')
-        el.classList.add('simplebar-content-wrapper')
-        return el
-      })(),
-      onExcludeGranule: jest.fn(),
-      onFocusedGranuleChange: jest.fn(),
-      onMetricsDataAccess: jest.fn()
-    }
-  }
+  // if (type === 'loading') {
```
Comment (trevorlang) [inline static/src/js/components/GranuleResults/__tests__/GranuleResultsList.test.js:32]: No longer needed

File: static/src/js/components/GranuleResults/__tests__/GranuleResultsBrowseImageCell.test.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+import { GranuleResultsBrowseImageCell } from '../GranuleResultsBrowseImageCell'
+
+Enzyme.configure({ adapter: new Adapter() })
+
+function setup(overrideProps) {
+  const props = {
+    row: {
+      original: {
+
```
Comment (abbottry) [inline static/src/js/components/GranuleResults/__tests__/GranuleResultsBrowseImageCell.test.js:13]: If this is supposed to be empty, remove the extra space here?

File: static/src/js/components/Panels/PanelGroupHeaderMeta.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+/**
+ * Renders PanelGroupHeaderMeta.
+ * @param {object} props - The props passed into the component.
+ * @param {string} props.children - Children components.
```
Comment (abbottry) [inline static/src/js/components/Panels/PanelGroupHeaderMeta.js:7]:
```suggestion
 * @param {String} props.children - Children components.
```

File: static/src/js/components/Panels/PanelGroupHeaderMeta.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+/**
+ * Renders PanelGroupHeaderMeta.
+ * @param {object} props - The props passed into the component.
```
Comment (abbottry) [inline static/src/js/components/Panels/PanelGroupHeaderMeta.js:6]:
```suggestion
 * @param {Object} props - The props passed into the component.
```

File: static/src/js/util/formatGranulesList.js [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+      producer_granule_id: producerGranuleId,
+      thumbnail: granuleThumbnail,
+      title: granuleTitle
+    } = granule
+
+    if (browseFlag && !hasBrowseImagery) hasBrowseImagery = true
+
+    const title = producerGranuleId || granuleTitle
+    const temporal = formattedTemporal
+    const timeStart = temporal[0]
```
Comment (abbottry) [inline static/src/js/util/formatGranulesList.js:45]:
```suggestion
    const [timeStart, timeEnd] = temporal
```

File: static/src/js/util/formatGranulesList.js [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+    const timeEnd = temporal[1]
```
Comment (abbottry) [inline static/src/js/util/formatGranulesList.js:46]:
```suggestion
```

File: static/src/js/components/GranuleResults/GranuleResultsHeader.js [hunk 13]
```diff
@@ -142,6 +153,7 @@ class GranuleResultsHeader extends Component {
     const handoffLinks = generateHandoffs(metadata, collectionSearch, mapProjection)
 
     const initialLoading = ((pageNum === 1 && isLoading) || (!isLoaded && !isLoading))
+    // eslint-disable-next-line no-unused-vars
```
Comment (macrouch) [inline static/src/js/components/GranuleResults/GranuleResultsHeader.js:156]: unused variable? if so remove it

File: static/src/js/components/GranuleResults/__tests__/GranuleResultsBody.test.js [hunk 14]
```diff
@@ -47,22 +93,154 @@ function setup(options = {
...
+      collectionId,
+      loadNextPage,
+      onExcludeGranule,
+      onFocusedGranuleChange,
+      onMetricsDataAccess
+    } = props
+
+    const resultsTable = enzymeWrapper.find(GranuleResultsTable)
+
+    // console.log('resultsTable', resultsTable.debug())
```
Comment (macrouch) [inline static/src/js/components/GranuleResults/__tests__/GranuleResultsBody.test.js:137]: remove

File: static/src/js/components/GranuleResults/__tests__/GranuleResultsBody.test.js [hunk 15]
```diff
@@ -72,4 +250,115 @@ describe('GranuleResultsBody component', () => {
...
+            },
+            456: {
+              title: '456'
+            }
+          }
+        }
+      }
+    })
+
+    // console.log('enzymeWrapper', enzymeWrapper.debug())
```
Comment (macrouch) [inline static/src/js/components/GranuleResults/__tests__/GranuleResultsBody.test.js:278]: remove

File: static/src/js/components/GranuleResults/GranuleResultsHeader.js (hunk 13 above)
Comment (trevorlang) [reply static/src/js/components/GranuleResults/GranuleResultsHeader.js:156]: Man. Youve got the eagle eye for these things.

**Pull request merged**